lib/ansible/modules/system/known_hosts.py

```diff
@@ -240,7 +243,7 @@ def search_for_host_key(module,host,key,hash_host,path,sshkeygen):
 
     sshkeygen_command.insert(1,'-H')
     rc,stdout,stderr=module.run_command(sshkeygen_command,check_rc=False)
-    if rc!=0: #something went wrong
+    if rc not in (0, 1) or stderr != '': #something went wrong
```

Comment:
**sigh** i really hate compensating for these types of bugs .... but I don't see we have much of a choice. tempted to add a warning "your ssh is broken".